opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -471,6 +473,32 @@ def detect(self) -> "Resource":
         )
 
 
+class HostResourceDetector(ResourceDetector):
+    """
+       The HostResourceDetector detects host resources using OpenTelemetry.
+       """
+
+    def detect(self) -> "Resource":
+        try:
+            resource = Resource.get_empty()
```

Comment:
What do you think about replacing the code in this try block with something smaller, e.g.

```
return Resource({
    HOST_NAME: socket.gethostname(),
    HOST_ARCH: platform.machine(),
})
```

Reply:
Yes , it better